src/EditorFeatures/Core/Workspaces/ITextBufferVisibilityTracker.cs

```diff
@@ -65,7 +66,7 @@ internal static class ITextBufferVisibilityTrackerExtensions
             try
             {
                 // Listen to when the active document changed so that we startup work on a document once it becomes visible.
-                var delayTask = Task.Delay(timeSpan, cancellationToken);
+                var delayTask = asyncListener.Delay(DelayTimeSpan.NonFocus, cancellationToken);
                 await Task.WhenAny(delayTask, visibilityChangedTaskSource.Task).ConfigureAwait(false);
```

Comment:
```suggestion
                await Task.WhenAny(delayTask, visibilityChangedTaskSource.Task).ConfigureAwait(true);
```

You also don't need the switch to main thread inside the `finally` block.

Reply:
good point!